File: framework/languages/java/e2e/SpotService/Shared/src/main/java/systems/zlink/e2e/spotservice/shared/ScenarioActorFactory.java
package systems.zlink.e2e.spotservice.shared;
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.CompletionStage;

import systems.zlink.framework.actors.ZLinkActor;
import systems.zlink.framework.actors.ZLinkActorContext;
import systems.zlink.framework.actors.ZLinkActorFactory;

public final class ScenarioActorFactory implements ZLinkActorFactory {
    @Override
    public CompletionStage<ZLinkActor> create(
        ZLinkActorContext context) {
        return CompletableFuture.completedFuture(new ScenarioActor(context));
    }
}

File: framework/languages/java/e2e-kotlin/SpotService/Server/Session/src/main/kotlin/systems/zlink/e2e/kotlin/spotservice/session/spots/ScenarioActorFactory.kt
package systems.zlink.e2e.kotlin.spotservice.session.spots

import systems.zlink.framework.actors.ZLinkActor
import systems.zlink.framework.actors.ZLinkActorContext
import systems.zlink.framework.kotlin.ZLinkSuspendingActorFactory

class ScenarioActorFactory : ZLinkSuspendingActorFactory() {
    override suspend fun createActor(context: ZLinkActorContext): ZLinkActor =
        ScenarioActor(context.actorId(), context)
}
